addons/point_of_sale/static/tests/tours/pos_cash_rounding_tour.js

```diff
@@ -0,0 +1,307 @@
+                ProductScreen.clickPayButton(),
+
+                PaymentScreen.totalIs("15.70"),
+                PaymentScreen.clickPaymentMethod("Cash", true, { remaining: "0.0" }),
+
+                PaymentScreen.clickInvoiceButton(),
+                PaymentScreen.clickValidate(),
+
+                ReceiptScreen.receiptAmountTotalIs("15.72"),
+                ReceiptScreen.receiptRoundingAmountIs("-0.02"),
+                ReceiptScreen.receiptToPayAmountIs("15.70"),
+                ReceiptScreen.receiptChangeAmountIsNotThere(),
+                ReceiptScreen.clickNextOrder(),
+
+                // Second order.
+                ProductScreen.addOrderline("random_product", "1"),
+                ProductScreen.clickPartnerButton(),
+                ProductScreen.clickCustomer("AAAAAA"),
+                ProductScreen.clickPayButton(),
+
+                PaymentScreen.totalIs("15.70"),
+                PaymentScreen.clickPaymentMethod("Bank"),
+                PaymentScreen.clickNumpad("."),
+                PaymentScreen.remainingIs("15.70"),
+                PaymentScreen.clickNumpad("6 7"),
+                PaymentScreen.fillPaymentLineAmountMobile("Bank", "0.67"),
+                PaymentScreen.remainingIs("15.03"),
+                PaymentScreen.clickPaymentMethod("Cash", true, { remaining: "0.0" }),
+
+                PaymentScreen.clickInvoiceButton(),
+                PaymentScreen.clickValidate(),
+
+                ReceiptScreen.receiptAmountTotalIs("15.72"),
+                ReceiptScreen.receiptRoundingAmountIsNotThere(),
+                ReceiptScreen.receiptToPayAmountIsNotThere(),
+                ReceiptScreen.receiptChangeAmountIsNotThere(),
+                ReceiptScreen.clickNextOrder(),
+            ].flat(),
+    });
+
+registry
+    .category("web_tour.tours")
+    .add("test_cash_rounding_down_add_invoice_line_not_only_round_cash_method", {
+        steps: () =>
+            [
+                Chrome.startPoS(),
+                Dialog.confirm("Open Register"),
+
+                ProductScreen.addOrderline("random_product", "1"),
+                ProductScreen.clickPartnerButton(),
+                ProductScreen.clickCustomer("AAAAAA"),
+                ProductScreen.clickPayButton(),
+
+                PaymentScreen.totalIs("15.70"),
+                PaymentScreen.clickPaymentMethod("Bank"),
+                PaymentScreen.clickNumpad("."),
+                PaymentScreen.remainingIs("15.70"),
+                PaymentScreen.clickNumpad("6 7"),
+                PaymentScreen.fillPaymentLineAmountMobile("Bank", "0.67"),
+                PaymentScreen.remainingIs("15.03"),
+                PaymentScreen.clickPaymentMethod("Cash", true, { remaining: "0.0" }),
+
+                PaymentScreen.clickInvoiceButton(),
+                PaymentScreen.clickValidate(),
+
+                ReceiptScreen.receiptAmountTotalIs("15.72"),
+                ReceiptScreen.receiptRoundingAmountIs("-0.05"),
```

Comment:
-0.05 is weid. it's the rounding precision. perhaps when adding the cash payment, instead of 15.00, it should suggest 15.05.

Reply:
@caburj Instead of computing the cash amount based on 15.03, maybe I should compute it based on 15.03 + 0.02 (the original rounding that round from 15.72 to 15.0).
However, that's a big corner case and I'm not even sure someone use a different cash rounding method than HALF-UP so I prefer to try that way first

Reply:
> Instead of computing the cash amount based on 15.03, maybe I should compute it based on 15.03 + 0.02 (the original rounding that round from 15.72 to 15.0).

This should be the generic behavior. We should always compute the "recommended" payment based on the remaining.

Config: HALF-UP, 0.05, all payment types
Order: Total amount to pay = 15.72

- Situation 1: single payment
- Whatever payment method is click, the automatically set payment amount is **15.70**.

- Situation 2: multiple payments
- Clicking Bank will automatically put: 15.70.
- Change it to 0.68 (the value isn't properly rounded which means the cashier is taking control, whatever they put, we should respect). This results to a remaining of 15.04.
- Clicking Cash should result to a payment line with amount = 15.05 (rounded from the remaining amount).

> that's a big corner case and I'm not even sure someone use a different cash rounding method than HALF-UP so I prefer to try that way first

In my opinion, multiple-line payments happen all the time, so I think we should consider it as a normal case. And it's good that you are adding these test cases.

Reply:
@caburj done (in a separated commit)